bfvmm/include/vmcs/vmcs_intel_x64.h

```diff
@@ -728,10 +758,22 @@ namespace guest_es_selector
         constexpr const auto name = "index";
 
         inline auto get()
-        { return (vm::read(addr, name) & mask) >> from; }
+        { return (get_vmcs_field(addr, name, exists()) & mask) >> from; }
+
+        inline auto get_if_exists(bool verbose = false) noexcept
+        { return (get_vmcs_field_if_exists(addr, name, verbose, exists()) & mask) >> from; }
 
         template<class T> void set(T val)
-        { vm::write(addr, (vm::read(addr, name) & ~mask) | ((val << from) & mask), name); }
+        {
+            auto field = get_vmcs_field(addr, name, exists());
```

Comment:
This is fine because it's an int (best optimization), but for future reference, please watch this, and if you find any issues with what we have done, please correct them. 

https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers